__tests__/catch-or-return.js

```diff
@@ -129,7 +144,28 @@ ruleTester.run('catch-or-return', rule, {
       errors: [{ message: catchMessage }]
     },
     {
-      code: 'frank.then(to).finally(fn)',
+      code: 'frank().then(to).catch(fn).then(foo)',
+      errors: [{ message: catchMessage }]
+    },
+    {
+      code: 'frank().finally(fn)',
+      errors: [{ message: catchMessage }]
+    },
```

Comment:
These two "invalid" examples are not directly related to this new option, but I thought they were useful to round out some rule-breaking cases that were not previously covered (a lone `finally` method, and a `then` after a `catch`)